src/uwtools/resources/rocoto.jsonschema

```diff
@@ -0,0 +1,72 @@
+{
+    "definitions": {
+        "dateTimePattern": {
+            "type": "string",
+            "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2}):(\\d{2})$"
+        }
+    },
+    "type": "object",
+    "properties": {
+        "workflow": {
+            "type": "object",
+            "properties": {
+                "attrs": {
+                    "type": "object",
+                    "properties": {
+                        "realtime": {
+                            "type": "string",
+                            "enum": [
+                                "true",
+                                "false",
+                                "T",
+                                "F",
+                                "t",
+                                "f",
+                                "True",
+                                "False",
+                                "FALSE",
+                                "TRUE"
+                            ]
+                        },
+                        "scheduler": {
+                            "type": "string",
+                            "enum": [
+                                "sge",
+                                "lsf",
+                                "lsfcray",
+                                "ll",
+                                "torque",
+                                "moabtorque",
+                                "moab",
+                                "pbspro",
+                                "slurm",
+                                "cobalt"
+                            ]
+                        },
+                        "cycylethrottle": {
+                            "type": "integer",
+                            "minimum": 0
+                        },
+                        "taskthrottle": {
+                            "type": "integer",
+                            "minimum": 0
+                        }
+                    },
+                    "required": ["realtime", "scheduler"]
+                },
+                "cycledefs": {
+                    "$ref": "#/definitions/dateTimePattern"
```

Comment:
Here is where we reference the dateTimePattern being used for the start, stop, increment method for the cycledef tag

Ex:
```
<cycledef>201101010000 201112311800 06:00:00</cycledef>
<cycledef>yyyymmddhhmm yyyymmddhhmm dd:hh:mm:ss</cycledef>
```

Reply:
This looks like it's headed in the right direction -- maybe it's even there. I just did some light googling to try to understand, and I found https://json-schema.org/understanding-json-schema/structuring.html#defs, which kind of sounds like it's come into play here. I see a `definitions` section like what you've got in this part of the docs https://json-schema.org/understanding-json-schema/structuring.html#bundling, but that seems to be within a `$defs` block. I really don't know, but this bit of text from the docs seems pertinent to our use case:

> Sometimes we have small subschemas that are only intended for use in the current schema and it doesn’t make sense to define them as separate schemas. Although we can identify any subschema using JSON Pointers or named anchors, the `$defs` keyword gives us a standardized place to keep subschemas intended for reuse in the current schema document.

Does that come into play here?

Reply:
I see, there are two different options here `$ref` and `$def`. 
`$ref` is used when we want to define reusable schemas that is used to reference an external schema elsewhere in our JSONSchema docs or external files.
`$def` is used when we want to define reusable schemas locally meaning within the same JSONSchema doc. 

I agree that here we will want to go with a `$def` block over a `$ref` block.

Reply:
I see, there are two different options here `$ref` and `$def`. 
`$ref` is used when we want to define reusable schemas that is used to reference an external schema elsewhere in our JSONSchema docs or external files.
`$def` is used when we want to define reusable schemas locally meaning within the same JSONSchema doc. 

I agree that here we will want to define a `$def` block to hold the reusable definitions.